### PR TITLE
Module for fetching metadata independent of ebrains

### PR DESCRIPTION
**Pull request by man-shu**
This PR intends to solve #4 
- The metadata file containing all information about IBC datasets on EBRAINS is hosted [here](https://github.com/individual-brain-charting/docs/blob/main/datasets.json)
- That file references .csv database files that are [here](https://github.com/individual-brain-charting/docs/tree/main)
- In the `metadata.py` we use curl and github API to fetch and store these files with the IBC API package
- In `utils.py` we now pass around a dictionary `METADATA` containing all required information to locate a dataset on EBRAINS

### CODE REVIEW

File: src/ibc_api/metadata.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+def _load_json(data_file):
+    """Read a given json file"""
+    with open(data_file, "r") as f:
+        data = json.load(f)
+
+    return data
+
+
+def ibc_subjects():
```
Comment (bthirion) [inline src/ibc_api/metadata.py]: Why do we need a function to define a constant ?

File: src/ibc_api/utils.py [hunk 2]
```diff
@@ -84,14 +75,14 @@ def _create_root_dir(dir_path=None):
...
+def get_info(data_type="volume_maps", save_to=None, metadata=METADATA):
     """Fetch a csv file describing each file in a given IBC dataset on EBRAINS.
 
     Parameters
     ----------
     data_type : str, optional
-        dataset to fetch, by default "statistic_map", TODO one of
-        ["statistic_map", "raw", "preprocessed"]
+        dataset to fetch, by default "volume_maps", one of
+        ["volume_maps","surface_maps", "raw", "preprocessed"]
```
Comment (bthirion) [inline src/ibc_api/utils.py]:
```suggestion
        ["volume_maps", "surface_maps", "raw", "preprocessed"]
```

**Review by bthirion — COMMENTED** [review]
Thx. LGTM.

File: src/ibc_api/metadata.py (hunk 1 above)
Comment (man-shu) [reply src/ibc_api/metadata.py]: Ok yes we don't! Will fix that. Thanks!

**Comment by bthirion** [comment]
Is the code functional ? (still in draft mode ...)

**Comment by man-shu** [comment]
Yes! Now it is! The `get_data.ipynb` and `example.py` are also up-to-date. You can refer to those for a walkthrough.

File: src/ibc_api/metadata.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+def _load_json(data_file):
+    """Read a given json file"""
+    with open(data_file, "r") as f:
+        data = json.load(f)
+
+    return data
+
+
+def select_dataset(data_type, metadata=None):
+    """Select metadata of the requested dataset"""
```
Comment (bthirion) [inline src/ibc_api/metadata.py]: if this meant to be partof the public API, include a complete doctring

File: src/ibc_api/metadata.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    return latest_version_index
+
+
+def fetch_remote_file(
+    file="datasets.json",
+    remote_root=REMOTE_ROOT,
+    local_root=LOCAL_ROOT,
+):
+    """Fetch a file from the IBC docs repo"""
```
Comment (bthirion) [inline src/ibc_api/metadata.py]: same here (alternatively, you can add a leading underscore to the function name)

**Review by bthirion — COMMENTED** [review]
The NB differences are sometimes cryptic.
LGTM overall. A small effort on docstrings would be welcome.

**Comment by man-shu** [comment]
I added the full docstrings. I think we are ready for merge @bthirion

**Review by bthirion — APPROVED** [review]
LGTM formally. It's a bit complicated for me to test the NB now, so I haven't done it. I trust you on that !

**Comment by man-shu** [comment]
Yes, @ferponcem tested it and it works. Merging then!

**Pull request merged**